src/Data/Functor/Coyoneda.hs

```diff
@@ -207,25 +211,75 @@ instance Distributive f => Distributive (Coyoneda f) where
   collect f = liftCoyoneda . collect (lowerCoyoneda . f)
   {-# INLINE collect #-}
 
+instance (Functor f, Show1 f) => Show1 (Coyoneda f) where
+#if LIFTED_FUNCTOR_CLASSES
+  liftShowsPrec sp sl d (Coyoneda f a) =
+    showsUnaryWith (liftShowsPrec sp sl) "liftCoyoneda" d (fmap f a)
+  {-# INLINE liftShowsPrec #-}
+#else
+  showsPrec1 d (Coyoneda f a) = showParen (d > 10) $
+    showString "liftCoyoneda " . showsPrec1 11 (fmap f a)
+  {-# INLINE showsPrec1 #-}
+#endif
+
+instance (Read1 f) => Read1 (Coyoneda f) where
+#if LIFTED_FUNCTOR_CLASSES
+  liftReadsPrec rp rl = readsData $
+    readsUnaryWith (liftReadsPrec rp rl) "liftCoyoneda" liftCoyoneda
+  {-# INLINE liftReadsPrec #-}
+#else
+  readsPrec1 d = readParen (d > 10) $ \r' ->
+    [ (liftCoyoneda f, t)
+    | ("liftCoyoneda", s) <- lex r'
+    , (f, t) <- readsPrec1 11 s
+    ]
+  {-# INLINE readsPrec1 #-}
+#endif
+
 instance (Functor f, Show (f a)) => Show (Coyoneda f a) where
   showsPrec d (Coyoneda f a) = showParen (d > 10) $
     showString "liftCoyoneda " . showsPrec 11 (fmap f a)
   {-# INLINE showsPrec #-}
 
-#ifdef __GLASGOW_HASKELL__
 instance Read (f a) => Read (Coyoneda f a) where
+#ifdef __GLASGOW_HASKELL__
   readPrec = parens $ prec 10 $ do
```

Comment:
FWIW, there _is_ `liftReadPrec` [in `base-4.10`](http://git.haskell.org/ghc.git/blob/a1ffd70835fba3aeb5be5c5d526e068cafe277be:/libraries/base/Data/Functor/Classes.hs#l161) (GHC 8.2). But that's not terribly important here.